transport/src/main/java/io/netty/channel/SimpleChannelEventHandler.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2013 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.channel;
+
+import io.netty.util.internal.TypeParameterMatcher;
+
+/**
+ * {@link ChannelInboundHandlerAdapter} which allows to conveniently only handle a specific type of user events.
+ *
+ * For example, here is an implementation which only handle {@link String} user events.
+ *
+ * <pre>
+ *     public class StringEventHandler extends
+ *             {@link SimpleChannelEventHandler}&lt;{@link String}&gt; {
+ *
+ *         {@code @Override}
+ *         protected void eventReceived({@link ChannelHandlerContext} ctx, {@link String} evt)
+ *                 throws {@link Exception} {
+ *             System.out.println(evt);
+ *         }
+ *     }
+ * </pre>
+ */
+public abstract class SimpleChannelEventHandler<I> extends ChannelInboundHandlerAdapter {
```

Comment:
`SimpleUserEventChannelHandler` (as user events are inbound all the time)?

Reply:
Neither name implies a tendency towards or against "inbound", do they?  But either way, I agree that name sounds better.  Changing.